datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -0,0 +1,76 @@
+package chaindatafetcher
+
+import (
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/log"
+	"github.com/klaytn/klaytn/networks/p2p"
+	"github.com/klaytn/klaytn/networks/rpc"
+	"github.com/klaytn/klaytn/node"
+)
+
+var logger = log.NewModuleLogger(log.ChainDataFetcher)
+
+type ChainDataFetcher struct {
+	config *ChainDataFetcherConfig
+
+	scheduler Scheduler
+	handlers  []ChainDataHandler
+}
+
+func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig) (*ChainDataFetcher, error) {
+	reqCh := make(chan Request, cfg.RequestChSize)
+	resCh := make(chan Response, cfg.ResponseChSize)
+
+	s := NewScheduler(cfg.BlockChannelSize, reqCh, resCh)
+
+	var h []ChainDataHandler
+	for i := 0; i < cfg.NumHandlers; i++ {
+		h = append(h, NewChainDataHandler(i, reqCh, resCh))
+	}
+
+	f := &ChainDataFetcher{
+		config:    cfg,
+		scheduler: s,
+		handlers:  h,
+	}
+	return f, nil
+}
+
+func (f *ChainDataFetcher) Protocols() []p2p.Protocol {
+	return []p2p.Protocol{}
+}
+
+func (f *ChainDataFetcher) APIs() []rpc.API {
+	return []rpc.API{}
+}
+
+func (f *ChainDataFetcher) Start(server p2p.Server) error {
+	for _, h := range f.handlers {
+		go h.Start()
+	}
+	f.scheduler.Start()
+	logger.Info("chaindatafetcher is started")
+	return nil
+}
+
+func (f *ChainDataFetcher) Stop() error {
+	f.scheduler.Stop()
+	for _, h := range f.handlers {
+		h.Stop()
+	}
+	logger.Info("chaindatafetcher is stopped")
+	return nil
```

Comment:
How about add a wait group to wait for scheduler and handlers to stop?

Reply:
Is a wait group necessary for all schedulers to be stopped? I think chaindata_handlers can be stopped in parallel. 6699a00f7a99a3e8a4102c62746154afda1381fd

Reply:
Yes, it is  `Service` interface in enroll as a EN service.